programs/clearing_house/src/lib.rs

```diff
@@ -1349,6 +1361,170 @@ pub mod clearing_house {
         Ok(())
     }
 
+    #[access_control(
+        exchange_not_paused(&ctx.accounts.state)
+    )]
+    pub fn settle_expired_market(ctx: Context<UpdateAMM>, market_index: u64) -> Result<()> {
+        let clock = Clock::get()?;
+        let now = clock.unix_timestamp;
+
+        let remaining_accounts_iter = &mut ctx.remaining_accounts.iter().peekable();
+        let mut oracle_map = OracleMap::load(remaining_accounts_iter, clock.slot)?;
+        let bank_map = BankMap::load(
+            &get_writable_banks(QUOTE_ASSET_BANK_INDEX),
+            remaining_accounts_iter,
+        )?;
+        let market_map = MarketMap::load(
+            &get_market_set(market_index),
+            &MarketSet::new(),
+            remaining_accounts_iter,
+        )?;
+
+        controller::repeg::update_amm(
+            market_index,
+            &market_map,
+            &mut oracle_map,
+            &ctx.accounts.state,
+            &clock,
+        )?;
+
+        let market = &mut market_map.get_ref_mut(&market_index)?;
+
+        validate!(
+            market.expiry_ts != 0,
+            ErrorCode::DefaultError,
+            "Market isn't set to expire"
+        )?;
+
+        validate!(
+            market.expiry_ts <= now,
+            ErrorCode::DefaultError,
+            "Market hasn't expired yet"
+        )?;
+
+        validate!(
+            market.amm.net_unsettled_lp_base_asset_amount == 0 && market.amm.user_lp_shares == 0,
+            ErrorCode::DefaultError,
+            "Outstanding LP in market"
+        )?;
+
+        let bank = &mut bank_map.get_ref_mut(&QUOTE_ASSET_BANK_INDEX)?;
+        let fee_reserved_for_protocol = cast_to_i128(market.amm.total_exchange_fee / 2)?;
+
+        let budget = market
+            .amm
+            .total_fee_minus_distributions
+            .checked_sub(fee_reserved_for_protocol)
+            .ok_or_else(math_error!())?
+            .max(0);
+
+        let available_fee_pool = cast_to_i128(get_token_amount(
+            market.amm.fee_pool.balance,
+            bank,
+            &BankBalanceType::Deposit,
+        )?)?
+        .checked_sub(fee_reserved_for_protocol)
+        .ok_or_else(math_error!())?
+        .max(0);
+
+        let fee_pool_transfer = budget.min(available_fee_pool);
+
+        controller::bank_balance::update_bank_balances(
+            fee_pool_transfer.unsigned_abs(),
+            &BankBalanceType::Borrow,
+            bank,
+            &mut market.amm.fee_pool,
```

Comment:
this is more reason to aggressively fill up fee pool (fee pool gets first claim on user negative pnl). otherwise you can end up in a situation where many users havent settled unrealized pnl and thus the fee pool isnt as full as it'd otherwise be